app/perfsymboltable.cpp

```diff
@@ -765,9 +771,8 @@ int PerfSymbolTable::lookupFrame(Dwarf_Addr ip, bool isKernel,
 
     Dwfl_Module *mod = module(ip, elf);
 
-    PerfUnwind::Location addressLocation(
-                (m_unwind->architecture() != PerfRegisterInfo::ARCH_ARM || (ip & 1))
-                ? ip : ip + 1, 0, -1, m_pid);
+    bool isArmArch = (m_unwind->architecture() == PerfRegisterInfo::ARCH_ARM);
+    PerfUnwind::Location addressLocation((!isArmArch || (ip & 1)) ? ip : ip + 1, 0, -1, m_pid);
```

Comment:
use alignedAddress here and any other places that use this pattern

Reply:
this task is still open

Reply:
still open

Reply:
Do you want me to enter one more utility function to align address?
alignedAddress is not appropriate for this case.

Reply:
```
(!isArmArch || (ip & 1)) ? ip : ip + 1
```
that is equivalent to
```
alignedAddress(ip + 1)
```
no?

generally - can you please add a comment to indicate why one is adding 1 and the other one is subtracting 1?

Reply:
Ok

Reply:
> ```
> (!isArmArch || (ip & 1)) ? ip : ip + 1
> ```
> 
> that is equivalent to
> 
> ```
> alignedAddress(ip + 1)
> ```
> 
> no?
 Looks like no.
> generally - can you please add a comment to indicate why one is adding 1 and the other one is subtracting 1?

Reply:
> > ```
> > (!isArmArch || (ip & 1)) ? ip : ip + 1
> > ```
> > 
> > 
> > that is equivalent to
> > ```
> > alignedAddress(ip + 1)
> > ```
> > 
> > 
> > no?
       Looks like no.
> > generally - can you please add a comment to indicate why one is adding 1 and the other one is subtracting 1?

Reply:
true, but why do we need to go into plus direction in one case and in minus direction in the other?!

Reply:
We should adjust it at first in one direction and then adjust it back.
Please look at the explanation here:
https://code.qt.io/cgit/qt-creator/perfparser.git/commit/?h=3.6&id=be7d6f47f4309270cf7d288db47580c7037eafb9

Reply:
perfect, please add such a comment to this code here too to make it easier to understand

ideally - introduce a `symbolAddress` function which does the +1 and keep the `alignedAddress` to do -1. You could then even share the implementation code of both in a third helper function which takes the direction (+1 / -1) as a parameter.

Reply:
Fixed